corgidrp/pump_trap_calibration.py

```diff
@@ -0,0 +1,3281 @@
+    (the capture cross section for holes and the energy level), trap densities
+    (i.e., how many traps per pixel for each kind of trap found), and
+    information about the capture time constant (for potential future
+    analysis).  This function only works as intended for the EMCCD with its
+    electrodes' paticular electric potential shape that will
+    be used for trap pumping on the Roman telescope.  It can find up to two
+    traps per sub-electrode location per pixel.  The function has an option to
+    save a preliminary output that takes a long time to create (the dictionary
+    called 'temps') and an option to load that in and start the function at
+    that point.
+
+    The frames in the input_dataset should be SCI full frames that:
+    - have had their bias subtracted 
+    - have been corrected for nonlinearity
+    - have been divided by EMGAIN
+
+    The following parameters from the II&T Trap pumping code are stored in the
+    object calibration file: 
+    trap_densities : list
+        A list of lists, where a list is provided for each type of trap.
+        The trap density for a trap type is the # of traps in a given 2-D bin
+        of E and tau divided by the total number of pixels in the image area.
+        The binning by default is fine enough to distinguish all trap types
+        found in the literarture. Only the bins that contain non-zero entries
+        are returned here.  Each trap-type list is of the following format:
+        [trap density, E, cs].
+        E is the central value of the E bin, and cs is the central value of the
+        cs bin. Stored in a hdu extention named 'trap_densities'
+    bad_fit_counter : int
+        Number of times trap_fit() provided fits of amplitude vs phase time
+        that were below fit_thresh over all schemes and temperature (which
+        would include any preliminary trap identifications that were rejected
+        because they weren't consistent across all schemes for sub-electrode
+        locations).
+    pre_sub_el_count : int
+        Number of times a trap was identified before filtering them through
+        sub-electrode location.  The counter is over all schemes and
+        temperatures.
+    unused_fit_data : int or None
+        Number of times traps were identified that were not matched up for
+        sub-electrode location determination. 
+    unused_temp_fit_data : int
+        Number of times traps were identified that did not get used in
+        identifying release time constant values across all temperatures
+    two_or_less_count : int
+        Number of traps that only appeared at 2 or fewer temperatures.
+    noncontinuous_count : int
+        Number of traps that appeared at a noncontinuous series of
+        temperatures.
+
+    Args:
+        input_dataset (corgi.drp.Dataset): The input dataset to be analyzed. The dataset should be a stack of trap-pumped frames, sorted according to temperature and pumping scheme.
```

Comment:
I suppose they don't have to be sorted since the dataset is sorted in the code according to temp and scheme?